components/speedreader/speedreader_url_loader.cc

```diff
@@ -326,8 +326,17 @@ void SpeedReaderURLLoader::CompleteSending() {
   state_ = State::kCompleted;
   // Call client's OnComplete() if |this|'s OnComplete() has already been
   // called.
-  if (complete_status_.has_value())
+  if (complete_status_.has_value()) {
     destination_url_loader_client_->OnComplete(complete_status_.value());
+    if (throttle_) {
+      // When distillation fails we return the original data. Causes this
+      // throttle to get deleted.
+      //
+      // TODO(iefremov): Is it okay to rely on just this, or should the distill
```

Comment:
I'm not sure, can you please point out when it gets deleted? Probably ok, but I can't quickly find a reference

Reply:
i verified heuristically, which is why i was asking. if the throttle returned the original data (distill failed) that function is never reached.